ui/apps/platform/src/Components/CompoundSearchFilter/utils/utils.tsx

```diff
@@ -105,6 +113,26 @@ export function ensureConditionNumber(value: unknown): { condition: string; numb
     };
 }
 
+export function ensureConditionDate(value: unknown): { condition: string; date: string } {
+    if (
+        typeof value === 'object' &&
+        value !== null &&
+        'condition' in value &&
+        'date' in value &&
+        typeof value.condition === 'string' &&
+        typeof value.date === 'string'
+    ) {
+        return {
+            condition: value.condition,
+            date: value.date,
+        };
+    }
+    return {
+        condition: dateConditions[1],
+        date: '',
+    };
+}
+
```

Comment:
Not a request for change, but there is a pattern I think works well for when
1. You have a value that you cannot be sure of the type at compile time
2. You want to parse that value into a safe type that you can rely on at compile and runtime, explicitly handling the case where the `unknown` value is not valid

This works by:
1. Instead of defining a TS type, describe the type as a `yup` schema
2. Derive the TS type from the schema
3. Use `yup`'s built in validation to parse the `unknown` type into a valid representation of your type, with a potential error fallback

```ts
const conditionDateSchema = yup.object({
    condition: yup.string().required(),
    date: yup.string().required(),
});

export type ConditionDate = yup.InferType<typeof conditionDateSchema>;

export function ensureConditionDate(value: unknown): ConditionDate {
    try {
        return conditionDateSchema.validateSync(value);
    } catch (error) {
        // Fallback in case of validation error
        return {
            condition: dateConditions[1], // Alternatively - return a string constant here
            date: '',
        };
    }
}
```

This is nice because it really doesn't add too much code (and in many cases is _less_ code) than doing the validation manually. It also has the benefit of avoiding missed edge cases when dealing with more complex object structures.

Reply:
Oh, interesting suggestion. I think this could be useful alternative. If we create some sort of `validateAndConvert` function (or a better name) that takes the value, the schema, and a fallback value, then we can reuse it for other cases. I guess the only thing is that we need to use `yup` here. Maybe we can consider this in an upcoming PR